### ISSUE TITLE
Add a sniff that detects incorrect arguments passed to `add_filter/add_action`

### ISSUE THREAD

**Issue by rinatkhaziev** [Automattic/VIP-Coding-Standards#265]
We need to make sure that only integers are passed as $priority, and $num_args.

E.g. `add_action( 'my_cool_hook, null, 'blarg' )` should trigger an error (or warning).

**Comment by GaryJones** [comment]
We had a similar request recently for something like this before.

How is this VIP-specific? 
Why check these specific functions for strict typing? Why not _every_ WP and PHP function? 

Static analysis would be **much** better suited to a task like this IMO, than sniffs.

Just one example:

```php
$priority = MY_WP_DEFAULT_PRIORITY; // Made up constant, defined elsewhere as 10.
add_action( 'my_cool_hook', 'blarg', $priority );
```

The code above will work just fine, but PHPCS wouldn't be able to categorically state whether the code is valid or not, since it only sees tokens, and not the value of the variable (and constant). The best it could give, is false positives = redundant noise.
